plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
I'm in favor of having both an open port check component and a more fuzzy 'Host Health' component, I believe both have their uses. It's useful to know if your Host is connectable from the outside world separately from if you are receiving settings calls. What if we add another component to this PR that conveys a 'health' metric based on the historic values of the RPC metrics?